runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -116,6 +117,55 @@ static uint32_t logBase2(uintptrj_t n)
    return result;
    }
 
+static TR_OpaqueMethodBlock *
+getMethodFromBCInfo(
+      TR_ByteCodeInfo &bcInfo,
+      TR::Compilation *comp)
+   {
+   TR_OpaqueMethodBlock *method = NULL;
+
+   if (bcInfo.getCallerIndex() >= 0)
+      method = (TR_OpaqueMethodBlock *)(comp->getInlinedCallSite(bcInfo.getCallerIndex())._vmMethodInfo);
```

Comment:
Curious why doesn't `getInlinedCallSite` just return `getCurrentMethod()` if index is -1?

Reply:
the `_inlinedCallSites` is an array of TR_InlinedCallSiteInfo which encapsulate info about a callsite including `TR_InlinedCallSite` and bytecode info and some other queries that's only meaningful for an actual call node. Creating a dummy callsite for the top most method seem to be convenient for this particular code but would probably confuse a lot of other code expecting an actual callsite.